src/algebra/char_zero.lean

```diff
@@ -5,25 +5,27 @@ Authors: Mario Carneiro
 
 Natural homomorphism from the natural numbers into a monoid with one.
 -/
-import data.nat.cast algebra.group algebra.field
+import data.nat.cast algebra.group algebra.field tactic.wlog
 
 /-- Typeclass for monoids with characteristic zero.
   (This is usually stated on fields but it makes sense for any additive monoid with 1.) -/
 class char_zero (α : Type*) [add_monoid α] [has_one α] : Prop :=
-(cast_inj : ∀ {m n : ℕ}, (m : α) = n ↔ m = n)
+(cast_injective : function.injective (coe : ℕ → α))
```

Comment:
If we're renaming the field, I guess you might as well call it `coe_injective`?
Also: good catch.

Reply:
I'm not sure what is the naming convention here: we have a lot of `nat.cast_*` lemmas.

Reply:
Hmmm, actually I also don't really know. I'm fine with leaving it like this for now.